dev/benchmarks/macrobenchmarks/lib/src/animated_complex_image_filtered.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+import 'package:flutter/material.dart';
+
+// Various tests to verify that Animated image filtered layers do not
+// dirty children even without explicit repaint boundaries. These intentionally use
+// text to ensure we don't measure the opacity peephole case.
+class AnimatedComplexImageFiltered extends StatefulWidget {
+  const AnimatedComplexImageFiltered({ super.key });
+
+  @override
+  State<AnimatedComplexImageFiltered> createState() => _AnimatedComplexImageFilteredState();
+}
+
+class _AnimatedComplexImageFilteredState extends State<AnimatedComplexImageFiltered> with SingleTickerProviderStateMixin {
+  late final AnimationController controller = AnimationController(vsync: this, duration: const Duration(milliseconds: 5000));
+  late final Animation<double> animation = controller.drive(Tween<double>(begin: 0.0, end: 1.0));
+  ui.ImageFilter imageFilter = ui.ImageFilter.blur();
+
+  @override
+  void initState() {
+    super.initState();
+    controller.forward(from: 0.0);
```

Comment:
The pageDelay is 1s and the duration is 5s so we lose 1s of this animation and it stops at 4s into the 5s duration.

Perhaps this should be executed on repeat so that it animates for the full duration of the benchmark?

Reply:
I can change the animation so that it repeats forever. that said, for benchmark duration I'd prefer to keep them consistent across benchmarks.

Reply:
In that case 5s is in the small minority of benchmark durations. The first animated_foo benchmark branched off from the 10s norm and others that have a similar name (even though they really have a similar structure to the others) have followed its lead rather than the more common 10s...

Reply:
Yeah makes sense -  updated the animated opacity, image_filtered, and placeholder tests as well